nexus/db-model/src/ip_pool.rs

```diff
@@ -103,15 +142,38 @@ impl IpPool {
             ),
             ip_version,
             rcgen: 0,
+            reservation_type: IpPoolReservationType::ExternalSilos,
+        }
+    }
+
+    /// Create a new pool reserved for Oxide's internal use.
+    pub fn new_oxide_internal(
```

Comment:
I'm torn between separate methods for different reservation types (which has the big plus of backwards compatibility with all the existing callers) and changing the existing functions to force them to pass in a reservation type for clarity. Maybe this could be `new_with_reservation_type(pool, ip_version, reservation_type)`, and then the implementations of `new()`, `new_v4()`, and `new_v6()` could just call it with specific args? Might be a reasonable middle ground. If there aren't very many callers of those, I'd also be tempted to remove this new method and instead add a `reservation_type` argument to `new`/`new_v4`/`new_v6`.